File: frontend/src/lib/workspace.ts
import { api } from "./api";

export interface WorkspaceSummary {
  counts: { datasets: number; connections: number; models: number; chats: number };
  recent_datasets: {
    id: string;
    name: string;
    n_rows: number;
    n_columns: number;
    source_type: string;
    created_at: string;
  }[];
  recent_models: {
    id: string;
    dataset_id: string;
    dataset_name: string;
    target: string;
    problem_type: string;
    best_model_name: string | null;
    created_at: string | null;
  }[];
  recent_queries: {
    id: string;
    connection_id: string;
    connection_name: string;
    question: string;
    created_at: string;
  }[];
}

export async function getWorkspaceSummary(): Promise<WorkspaceSummary> {
  const { data } = await api.get<WorkspaceSummary>("/workspace/summary");
  return data;
}

File: backend/app/schemas/workspace.py
"""Pydantic schemas for the analytics workspace summary."""
import uuid
from datetime import datetime

from pydantic import BaseModel


class WorkspaceCounts(BaseModel):
    datasets: int
    connections: int
    models: int
    chats: int


class RecentDataset(BaseModel):
    id: uuid.UUID
    name: str
    n_rows: int
    n_columns: int
    source_type: str
    created_at: datetime


class RecentModel(BaseModel):
    id: uuid.UUID
    dataset_id: uuid.UUID
    dataset_name: str
    target: str
    problem_type: str
    best_model_name: str | None
    created_at: datetime | None


class RecentQuery(BaseModel):
    id: uuid.UUID
    connection_id: uuid.UUID
    connection_name: str
    question: str
    created_at: datetime


class WorkspaceSummary(BaseModel):
    counts: WorkspaceCounts
    recent_datasets: list[RecentDataset]
    recent_models: list[RecentModel]
    recent_queries: list[RecentQuery]
